pandas/core/arrays/sparse/array.py

```diff
@@ -372,8 +375,11 @@ def __init__(
 
     @classmethod
     def _simple_new(
-        cls, sparse_array: np.ndarray, sparse_index: SparseIndex, dtype: SparseDtype
-    ) -> ABCSparseArray:
+        cls: Type[_SparseArrayT],
```

Comment:
needing to annotate `cls` seems weird.  is that just me?

Reply:
annotated for generic self. see https://mypy.readthedocs.io/en/latest/generics.html#generic-methods-and-generic-self

again could see how far we get without this.